Add dedicated "add style" button

Sjoerd1993 opened the pull request:
Instead of having a copy button on each style row, this adds one single "add style" button that pops up a dialog. It then gives a dropdown with a choice which style to base the new one on, and a field for the name of the new style. The name is automatically filled in based on the template the user chooses, but can be changed manually of course .

Other than this, there's not really something I really want in the upcoming release. Except for a warning dialog when deleting styles ("This will delete {style name}, are you sure") and for resetting the styles to the default settings, as those actions are destructive. See PR #229. 

![image](https://user-images.githubusercontent.com/68477016/232209183-672d1e01-abb9-4971-98cc-0099bd9ec107.png)

Sjoerd1993 commented:
This also cleans up the style list somewhat:

![image](https://user-images.githubusercontent.com/68477016/232209207-fffcf641-7508-4b13-9870-5890de33e139.png)

cmkohnen commented on src/plot_styles.py:
```diff
@@ -483,3 +490,41 @@ def __init__(self, parent, color):
+        self.confirm_button.connect("clicked", self.on_confirm)
+        self.plot_style_templates.connect("notify::selected",
+                                          self.on_template_changed)
+        self.set_transient_for(parent.parent.plot_styles_window)
```
parent would suffice here

cmkohnen commented on src/plot_styles.py:
```diff
@@ -389,22 +391,29 @@ def delete_style(self, _, style):
 
+    def add_data(self, _):
+        win = AddStyleWindow(self)
+        win.present()
```
assignment and present() is redundant here

cmkohnen commented on src/actions.py:
```diff
@@ -104,7 +104,7 @@ def export_figure_action(_action, _target, self):
 
 def plot_styles_action(_action, _target, self):
-    PlotStylesWindow(self)
+    self.plot_styles_window = PlotStylesWindow(self)
```
together with my comment on L513 this would be redundant

cmkohnen commented on data/meson.build:
```diff
@@ -58,6 +58,7 @@ endif
 
 blueprints = custom_target('blueprints',
   input: files(
+    'ui/add_style.blp',
```
might not be scope of this pr, but perhaps sort this list alphabetically (Probably oversight on my end)

cmkohnen commented on data/ui/style_box.blp:
```diff
@@ -19,16 +19,6 @@ template StyleBox : Box {
-    }
-  }
-
   Button delete_button {
```
Perhaps not in scope either, but maybe we should switch the delete and edit button for consistency

Sjoerd1993 commented:
Removed the cancel button and changed the margins to match the rest of the UI (e.g. Add Equation window): 

![image](https://user-images.githubusercontent.com/68477016/232225114-ac1fef9e-93d3-4bb4-ac46-3518669d4d47.png)

cmkohnen reviewed (commented):
I feel this is a good change
